Source/Core/Core/PowerPC/JitArm64/JitArm64_Integer.cpp

```diff
@@ -1274,23 +1267,29 @@ void JitArm64::subfex(UGeckoInstruction inst)
   else
   {
     gpr.BindToRegister(d, d == a || d == b);
-    ARM64Reg RB = mex ? gpr.GetReg() : gpr.R(b);
-    if (mex)
-      MOVI2R(RB, -1);
-
-    if (js.carryFlag == CarryFlag::ConstantTrue)
-    {
-      CARRY_IF_NEEDED(SUB, SUBS, gpr.R(d), RB, gpr.R(a));
-    }
-    else
     {
-      LoadCarry();
-      CARRY_IF_NEEDED(SBC, SBCS, gpr.R(d), RB, gpr.R(a));
+      auto RB = [&]() -> Arm64GPRCache::ScopedARM64Reg {
```

Comment:
Using a lambda for this makes the code harder to read than the old approach in my opinion. Using the old approach does mean you have to declare `RB` as `Arm64GPRCache::ScopedARM64Reg` instead of `auto`, but you have to write that out in the lambda return type here anyway.